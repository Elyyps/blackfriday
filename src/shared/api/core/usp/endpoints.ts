import { uniqueSellingPoints } from "./generate-dummy-data";
import { USP } from "./usp";

export const generateUSP = (): USP => uniqueSellingPoints;
