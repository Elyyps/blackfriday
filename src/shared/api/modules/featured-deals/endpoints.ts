import { dummyFeaturedDealsData } from "./generate-dummy-data";

export const getFeaturedDealsDummyData = () => dummyFeaturedDealsData;
