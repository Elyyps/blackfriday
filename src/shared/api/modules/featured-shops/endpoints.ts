import { dummyFeaturedShopsData } from "./generate-dummy-data";

export const getFeaturedShopsDummyData = () => dummyFeaturedShopsData;
