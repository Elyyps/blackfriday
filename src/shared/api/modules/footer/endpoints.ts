import { bottomFooter, middleFooter, topFooter } from "./generate-dummy-data";
import { Footer } from "./models/footer.module";

export const getFullfooter = (): Footer => ({
  bottomFooter: bottomFooter(),
  middleFooter: middleFooter(),
  topFooter: topFooter()
});
