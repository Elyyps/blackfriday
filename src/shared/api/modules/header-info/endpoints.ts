import { HeaderInfoDummy } from "./generate-dummy-data";

export const getHeaderInfoDummyData = () => HeaderInfoDummy;
