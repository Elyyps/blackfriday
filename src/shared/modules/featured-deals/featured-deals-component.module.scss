.featured-deals {
  box-sizing: content-box;
  max-width: 1200px;
  margin-left: auto;
  margin-right: auto;
  padding: 16px;

  &__list {
    display: flex;
    box-sizing: border-box;
    flex-wrap: wrap;
    align-items: center;
    justify-content: space-around;
    > * {
      width: 100%;
      flex: 0 1 19%;
      @media (max-width: 1139px) {
        flex: 0 1 24%;
      }
      @media (max-width: 914px) {
        flex: 0 1 32%;
      }
      @media (max-width: 699px) {
        width: 100%;
        padding: 4px 0px;
        flex: 0 1 100%;
      }
    }
  }
}

@media (max-width: 1139px) {
  .featured-deals {
    &__item-3 {
      display: none;
    }
  }
}
@media (max-width: 914px) {
  .featured-deals {
    &__item-2 {
      display: none;
    }
  }
}
