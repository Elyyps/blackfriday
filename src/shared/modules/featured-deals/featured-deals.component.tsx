import * as React from "react";
import styles from "./featured-deals-component.module.scss";
import { SeeMoreCard } from "@app/api/core/see-more-card";
import { SeeMoreCardComponent } from "@app/core/see-more-card";
import { DealCard } from "@app/api/core/deal-card";
import { DealCardComponent } from "@app/core/deal-card";

export interface IFeaturedDealsComponentProps {
  deals: [DealCard, DealCard, DealCard, DealCard];
  seeMoreCard: SeeMoreCard;
  title: string;
}

const FeaturedDealsComponent = (props: IFeaturedDealsComponentProps) => {
  const { deals, title, seeMoreCard } = props;

  return (
    <div className={styles["featured-deals"]}>
      <h2 className={styles["featured-deals__title"]}>{title}</h2>
      <div className={styles["featured-deals__list"]}>
        {deals.map((deal, key) => (
          <div className={styles[`featured-deals__item-${key}`]} key={key}>
            <DealCardComponent
              key={key}
              label={deal.label}
              subtitle={deal.subtitle}
              buttonLink={deal.button.url}
              title={deal.title}
              oldPrice={deal.oldPrice}
              image={deal.picture}
              buttonText={deal.button.title}
              newPrice={deal.newPrice}
              sale={deal.discountText}
            />
          </div>
        ))}
        <div>
          <SeeMoreCardComponent title={seeMoreCard.title} link={seeMoreCard.link} icon={seeMoreCard.icon} />
        </div>
      </div>
    </div>
  );
};

export { FeaturedDealsComponent };
