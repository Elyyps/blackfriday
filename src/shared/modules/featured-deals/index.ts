export * from "./featured-deals.component";
