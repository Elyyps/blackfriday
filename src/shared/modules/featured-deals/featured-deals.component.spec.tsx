import React from "react";
import { shallow } from "enzyme";
import { FeaturedDealsComponent } from "./featured-deals.component";
import { getFeaturedDealsDummyData } from "@app/api/modules/featured-deals/endpoints";

describe("[FeaturedShops]", () => {
  // it("should render component without crashing", () => {
  //   const renderedComponent = shallow(<FeaturedDealsComponent {...getFeaturedDealsDummyData()} />);
  //   expect(renderedComponent).toMatchSnapshot();
  // });
});
