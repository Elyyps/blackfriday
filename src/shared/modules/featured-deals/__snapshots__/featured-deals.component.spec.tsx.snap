// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[FeaturedShops] should render component without crashing 1`] = `
<div
  className="featured-deals"
>
  <h2
    className="featured-deals__title"
  >
    Meer deals
  </h2>
  <div
    className="featured-deals__list"
  >
    <div
      className="featured-deals__item-0"
      key="0"
    >
      <DealCardComponent
        buttonLink="/"
        buttonText="Naar deal"
        image="https://d2ddoduugvun08.cloudfront.net/items/2q1D2t2D003Z1X023z18/Image%202019-07-01%20at%204.45.18%20PM.png?X-CloudApp-Visitor-Id=3342233&v=d96f45c6"
        key="0"
        newPrice="€2.429"
        oldPrice="€3.429"
        sale="Bespaar 40%"
        subtitle="Mediamarkt"
        title="Pocketveermatas Pocket Comfort x2000"
      />
    </div>
    <div
      className="featured-deals__item-1"
      key="1"
    >
      <DealCardComponent
        buttonLink="/"
        buttonText="Naar deal"
        image="https://d2ddoduugvun08.cloudfront.net/items/42013C072O2711070D0a/Image%202019-07-01%20at%204.46.51%20PM.png?X-CloudApp-Visitor-Id=3342233&v=c0749925"
        key="1"
        label="Hot"
        newPrice="€2.429"
        oldPrice="€3.429"
        sale="Bespaar 40%"
        subtitle="Mediamarkt"
        title="Hoofdkussen Actie kussenset"
      />
    </div>
    <div
      className="featured-deals__item-2"
      key="2"
    >
      <DealCardComponent
        buttonLink="/"
        buttonText="Naar deal"
        image="https://d2ddoduugvun08.cloudfront.net/items/193s25422A3q0v2F1f35/Image%202019-07-01%20at%204.47.42%20PM.png?X-CloudApp-Visitor-Id=3342233&v=b6e3280e"
        key="2"
        newPrice="€2.429"
        oldPrice="€3.429"
        sale="Bespaar 40%"
        subtitle="Mediamarkt"
        title="Lattenbodem FLEXUS 2600 VL"
      />
    </div>
    <div
      className="featured-deals__item-3"
      key="3"
    >
      <DealCardComponent
        buttonLink="/"
        buttonText="Naar deal"
        image="https://d2ddoduugvun08.cloudfront.net/items/2R2d2B0a0e2o3528123s/Image%202019-07-01%20at%204.48.22%20PM.png?X-CloudApp-Visitor-Id=3342233&v=88efdc5a"
        key="3"
        newPrice="€2.429"
        oldPrice="€3.429"
        sale="Bespaar 40%"
        subtitle="Mediamarkt"
        title="Dyson apparaten met korting"
      />
    </div>
    <div>
      <SeeMoreCardComponent
        icon="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/footerIcons/store.svg"
        link={
          Object {
            "title": "Alle winkels",
            "url": "/",
          }
        }
        title="Wij hebben meer winkels in ons overzicht"
      />
    </div>
  </div>
</div>
`;
